File: united_chemie/public/js/production_plan.js
frappe.ui.form.on("Production Plan Item", {
    item_code: function(frm, cdt, cdn) {
        set_actual_stock(frm, locals[cdt][cdn]);
    },

    warehouse: function(frm, cdt, cdn) {
        set_actual_stock(frm, locals[cdt][cdn]);
    }
});


function set_actual_stock(frm, row) {

    if (!row.item_code || !row.warehouse) {
        frappe.model.set_value(
            row.doctype,
            row.name,
            "actual_stock",
            0
        );
        return;
    }

    frappe.db.get_value(
        "Bin",
        {
            item_code: row.item_code,
            warehouse: row.warehouse
        },
        "actual_qty"
    ).then(r => {

        let actual_qty = 0;

        if (r.message) {
            actual_qty = r.message.actual_qty || 0;
        }

        frappe.model.set_value(
            row.doctype,
            row.name,
            "actual_qty",
            actual_qty
        );
    });
}

File: united_chemie/united_chemie/doc_events/production_plan.py
import frappe

def validate(self, method):
    set_actual_stock(self, method)

def set_actual_stock(doc, method):
    for row in doc.po_items:
        if not row.item_code or not row.warehouse:
            row.actual_stock = 0
            continue

        actual_qty = frappe.db.get_value(
            "Bin",
            {
                "item_code": row.item_code,
                "warehouse": row.warehouse
            },
            "actual_qty"
        )

        row.actual_qty = actual_qty or 0